css-technical-plan.md

```diff
@@ -0,0 +1,131 @@
+# Atlas (CSS) tech plan
+
+## Links
+
+- [Current version of tokens](https://www.figma.com/proto/03HGueCJEx4tyZd3Rhfw5B/DevRel-Design-System?node-id=268%3A343&viewport=1143%2C-507%2C0.08135364204645157&scaling=scale-down-width)
+- [Colors](https://www.figma.com/file/6ZQIcp2AS8022uU96v0xx4/ATLAS---COLOR?node-id=0%3A1)
+
+## Task list
+
+👨‍💻 Constitutes a developer task
+💭 Pondering, not ready, but input welcome
+⚜ Group or PM task
+
+- fill out all Open Source related files ⚜
+  - update project README.md ⚜
+  - 👨‍💻 write initial setup steps
+  - 👨‍💻 write steps for inclusion in a project
+  - 👨‍💻 update css folder README.md
+  - ⚜ review and update CODE_OF_CONDUCT.md (group task / PM task?)
+- 👨‍💻 add monorepo related files
+  - add top level index file scss in /css
+  - add top level package.json and init monorepo
+- 👨‍💻 implment changesets [changesets](https://github.com/atlassian/changesets)
+- 👨‍💻 implement tokens (each direct sub list item is a dev task)
+  - color tokens
+    - convert tokens to hsla (pending dropping IE theming confirmation)
+    - raw colors in palette.scss
+    - themed colors in color-themes.scss
+  - 👨‍💻 spacing tokens
+  - 👨‍💻 typography, font-size tokens
+  - 👨‍💻 depth/shadow tokens
+  - 👨‍💻 write tokens readme
+- 👨‍💻 implement code formatting and linting
+  - requirements:
+    - code format on save
+    - fast via cli
+    - pre-commit hook w/husky
+  - proposed combination: prettier and stylelint combo, favoring airbnb stylelint presets
+- 👨‍💻 implment mixins folder
+  - first move media query mixins
+  - wait until we need a mixin before moving something else
+  - catalog shared and shareable mixins from docs-ui
+- implement Core folder (each sub item is a dev task)
+  - 👨‍💻 resets (normalize and minireset + custom?)
+  - 👨‍💻 create css custom properties for theme colors
+  - 👨‍💻 implement font stack
+    - no font face rules
+    - just native font family (from github / bulma)
+    - this will be overridden downstream
+  - 👨‍💻 implement focus styling (focus-visible) 💭
+    - escape hatch, which can toggle :focus-visible, rather than the polyfill (.focus-visible)
+  - 👨‍💻 implement a css build for prototyping 💭
+    - should include a font face?
+- devops setup
```

Comment:
Is the site in scope for this, or is it strictly the CSS for Atlas?

Reply:
Not for this plan, this is just the init of the css